source/extensions/common/aws/signers/iam_roles_anywhere_sigv4_signer.cc

```diff
@@ -0,0 +1,85 @@
+#include "source/extensions/common/aws/signers/iam_roles_anywhere_sigv4_signer.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Aws {
+
+std::string
+IAMRolesAnywhereSigV4Signer::createCredentialScope(absl::string_view short_date,
+                                                   absl::string_view override_region) const {
+  return fmt::format(IAMRolesAnywhereSigV4SignatureConstants::SigV4CredentialScopeFormat,
+                     short_date, override_region.empty() ? region_ : override_region,
+                     service_name_);
+}
+
+std::string IAMRolesAnywhereSigV4Signer::createStringToSign(
+    const X509Credentials x509_credentials, const absl::string_view canonical_request,
+    const absl::string_view long_date, const absl::string_view credential_scope) const {
+  auto& crypto_util = Envoy::Common::Crypto::UtilitySingleton::get();
+  if (x509_credentials.publicKeySignatureAlgorithm() ==
+      X509Credentials::PublicKeySignatureAlgorithm::RSA) {
+    return fmt::format(
+        IAMRolesAnywhereSigV4SignatureConstants::SigV4StringToSignFormat,
+        IAMRolesAnywhereSigV4SignatureConstants::X509SigV4RSA, long_date, credential_scope,
+        Hex::encode(crypto_util.getSha256Digest(Buffer::OwnedImpl(canonical_request))));
+  } else {
+    return fmt::format(
+        IAMRolesAnywhereSigV4SignatureConstants::SigV4StringToSignFormat,
+        IAMRolesAnywhereSigV4SignatureConstants::X509SigV4ECDSA, long_date, credential_scope,
+        Hex::encode(crypto_util.getSha256Digest(Buffer::OwnedImpl(canonical_request))));
+  }
+}
+
+std::string
+IAMRolesAnywhereSigV4Signer::createSignature(const X509Credentials x509_credentials,
+                                             const absl::string_view string_to_sign) const {
+
+  std::string key = x509_credentials.certificatePrivateKey().value();
+  auto keysize = x509_credentials.certificatePrivateKey().value().size();
+  bssl::UniquePtr<BIO> bio(BIO_new_mem_buf(key.c_str(), keysize));
+
+  bssl::UniquePtr<EVP_PKEY> pkey(PEM_read_bio_PrivateKey(bio.get(), nullptr, nullptr, nullptr));
```

Comment:
Seeing the usage of X509Credentials, it makes me wonder if it'd be better to keep around the private key as it's BoringSSL object in the credentials object, such that it doesn't have to reload them on every signing.